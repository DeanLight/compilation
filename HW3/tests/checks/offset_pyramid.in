byte foo(int x,int y,byte z)
{
	int w; //0
	{
		int r; //1
		{
			byte bb = 54b; //2 
			{
				bool bb2 = false; //3
			}
			bool bb2 = true; //3
		}
		int bb = 0; //2
	}
	byte r = z; //1
	x = y;
	return r;
}
void main()
{
	return;
}