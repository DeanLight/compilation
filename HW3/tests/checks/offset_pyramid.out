---end scope---
bb2 BOOL 3
---end scope---
bb BYTE 2
bb2 BOOL 3
---end scope---
r INT 1
bb INT 2
---end scope---
x INT -1
y INT -2
z BYTE -3
w INT 0
r BYTE 1
---end scope---
---end scope---
print (STRING)->VOID 0
printi (INT)->VOID 0
foo (INT,INT,BYTE)->BYTE 0
main ()->VOID 0
